src/bookshelf/src/bksw_wm_object.erl

```diff
@@ -122,7 +122,14 @@ generate_etag(Rq0, Ctx) ->
 
 delete_resource(Rq0, Ctx) ->
     {ok, Bucket, Path} = bksw_util:get_object_and_bucket(Rq0),
-    {bksw_io:entry_delete(Bucket, Path), Rq0, Ctx}.
+    Res = bksw_io:entry_delete(Bucket, Path),
+    case Res of
+        true ->
+            sync_deletion(bksw_io_names:entry_path(Bucket, Path)),
```

Comment:
Given that these functions themselves are just one-line pass through calls, maybe we should just call bksw_sync directly?